infrastructure/api/sse-api.yml

```
@@ -317,10 +318,47 @@ Resources:
     Properties:
       Handler: backend/api/src/handlers/auth/update-client.updateClientInRegistryHandler
       Description: Updates a client using an HTTP endpoint
+      KmsKeyArn: !Ref AuthKmsKey
       Environment:
         Variables:
           AUTH_REGISTRATION_BASE_URL: !Ref AuthRegistrationBaseUrl
           AUTH_API_KEY: "{{resolve:ssm:/self-service/api/auth-api-key}}"
+      Policies:
+        - Statement:
+            - Effect: Allow
+              Action:
+                - "kms:Decrypt"
+                - "kms:Encrypt"
+              Resource:
+                - !Ref AuthKmsKey
+  
+ # KMS configuration for auth kms key
+  AuthKmsKey:
```

```suggestion
 EnvironmentEncryptionKey:
```
This is used by our Lambdas to encrypt their environment, this name would probably be clearer

---

fixed